### PR TITLE
RCE content theme switcher 🎄

### PR DESCRIPTION
**Pull request by idevainstr**
refs: MBL-16352
affects: Student, Teacher, Parent
release note: Added an appearance switcher in dark mode for html content.
test plan: See ticket.

## Screenshots

<table>
<tr><th>Dark</th><th>Light</th></tr>
<tr>
<td></td>
<td></td>
</tr>
</table>

## Checklist

- [x] Follow-up e2e test ticket created or not needed
- [x] A11y checked
- [x] Tested on phone
- [ ] Tested on tablet
- [ ] Tested in dark mode
- [ ] Tested in light mode
- [ ] Approve from product or not needed

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Parent Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<details><summary>Teacher Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>



<details><summary>:x: XCTest failed: StudentUnitTests/QuizDetailsViewControllerTests/testLayout</summary>


    Unable to activate constraint with anchors <NSLayoutXAxisAnchor:0x600000042440 "Student.CoreWebView:0x7feb77a12200.leading"> and <NSLayoutXAxisAnchor:0x600000042140 "UIView:0x7feb3729f7f0.leading"> because they have no common ancestor.  Does the constraint or its anchors reference items in different view hierarchies?  That's illegal. (NSGenericException)
</details>
<details><summary>:x: XCTest failed: StudentUnitTests/QuizDetailsViewControllerTests/testNoQuiz</summary>


    Unable to activate constraint with anchors <NSLayoutXAxisAnchor:0x600000044440 "Student.CoreWebView:0x7feb67198800.leading"> and <NSLayoutXAxisAnchor:0x600000072bc0 "UIView:0x7feb370d0170.leading"> because they have no common ancestor.  Does the constraint or its anchors reference items in different view hierarchies?  That's illegal. (NSGenericException)
</details>


[MBL-16352](https://instructure.atlassian.net/browse/MBL-16352)

#### Affected Apps: Student, Teacher, Parent

#### Release Note: 
Added an appearance switcher in dark mode for html content.
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Review by szabinst — CHANGES_REQUESTED** [review]
Not sure if we can fix it but when you change modes, the profile image reloads everytime. 
https://user-images.githubusercontent.com/110813321/207893291-c2b85ad8-8358-4f69-b139-28f420f92ef1.MP4

This one is an edge case, but if you open a page with dark mode, and switch the system theme to light mode then part of the html content is not visible. 
https://user-images.githubusercontent.com/110813321/207893467-9b7263ce-94e6-4775-b929-650f54a00624.MP4

File: Core/Core/UIViews/CoreWebView.swift [hunk 1]
```diff
@@ -710,3 +739,103 @@ extension CoreWebView {
             .replacingOccurrences(of: ">", with: "&gt;")
     }
 }
+
+extension CoreWebView {
```
Comment (szabinst) [inline Core/Core/UIViews/CoreWebView.swift:743]: Since this file is already pretty large maybe we could extract it into an extension file.

**Review by szabinst — DISMISSED** [review]
looks good to me! I had remarks about a few improvement possibilities

**Review by vargaat — CHANGES_REQUESTED** [review]
- [x] Theme switcher is missing on the Teacher app's assignment details.
- [ ] The switch is also missing from the speedgrader (tested with a text submission).
- [x] Discussion reply screen's content size is incorrectly calculated and the message is cut off.

**Review by balintbartok — APPROVED** [review]
QA + 1

**Review by szabinst — APPROVED** [review]
QA+1

**Review by balintbartok — APPROVED** [review]
QA + 1

**Review by szabinst — APPROVED** [review]

**Pull request merged**